omisego-sdk/src/main/java/co/omisego/omisego/websocket/interval/SocketHeartbeat.kt

```diff
@@ -0,0 +1,59 @@
+package co.omisego.omisego.websocket.interval
+
+/*
+ * OmiseGO
+ *
+ * Created by Phuchit Sirimongkolsathien on 3/5/2018 AD.
+ * Copyright © 2017-2018 OmiseGO. All rights reserved.
+ */
+
+import co.omisego.omisego.model.socket.SocketSend
+import co.omisego.omisego.websocket.SocketClientContract
+import co.omisego.omisego.websocket.channel.SocketChannelContract
+import co.omisego.omisego.websocket.channel.SocketMessageRef
+import co.omisego.omisego.websocket.enum.SocketEventSend
+import java.util.Date
+import java.util.Timer
+import kotlin.concurrent.schedule
+
+class SocketHeartbeat(
+    /**
+     * [SocketMessageRef] is responsible for creating a unique ref value to be sent with the [SocketSend].
+     */
+    override val socketMessageRef: SocketChannelContract.MessageRef
+) : SocketClientContract.SocketInterval {
+    /**
+     * The timer for scheduling the [SocketSend] periodically to be sent to the server.
+     */
+    override var timer: Timer? = null
+
+    /**
+     * An interval of milliseconds between the end of the previous task and the start of the next one.
+     */
+    override var period: Long = 5000
+
+    /**
+     * Start to schedule the [SocketSend] to be sent to the server periodically.
+     *
+     * @param task A lambda with a [SocketSend] parameter. This will be executed periodically that starts immediately.
+     */
+    @Suppress("OVERRIDE_BY_INLINE")
+    override inline fun startInterval(crossinline task: (SocketSend) -> Unit) {
+        timer = Timer()
```

Comment:
Very dangerous, if the previous timer wasn't cancelled, you'll basically have a "phantom" thread hanging somewhere because you'll have dropped the reference.

Reply:
Plus, you probably need to have some sync block around this to prevent threading issues

Reply:
I think this commit https://github.com/omisego/android-sdk/pull/24/commits/47d47924d7a5847ab6a252b41ce617cfb454be77 will fix the issue.

Reply:
This commit https://github.com/omisego/android-sdk/pull/24/commits/7b0129fe4031f89cdef41f6f4fee54252d6868a7 explain the test assertion.